main.go

```diff
@@ -65,23 +66,23 @@ func main() {
 		return
 	}
 
-	log := logrus.New()
```

Comment:
To avoid passing a log instance and to avoid checking that the instance is not nil.
If we allow passing a pointer to logger, that value can be nil so, we need to check that the instance is not nil or use a constructor to check if the instance is nil and in that case, we should set it to `logrus.New`.